### ISSUE TITLE
Type pattern: definite assignment false positive for struct that has only reference type properties and is declared in another assembly

### ISSUE THREAD

**Issue by HellBrick** [dotnet/roslyn#38723]
**Version Used**: 
Visual Studio 16.2.5
MSBuild 16.2.37902+b5aaefc9f
csc.exe 3.2.1-beta4-19408-03

**Steps to Reproduce**:
You need two assemblies for this, `Lib` and `App`, where `App` references `Lib`.
```csharp
// Lib
namespace Lib
{
	public struct StructWithStringProp
	{
		public string Text { get; set; }
	}
}

// App
namespace App
{
	internal class CallSites
	{
		private static string ExternalStringPropCallSite(Lib.StructWithStringProp? nullable)
		{
			bool flag = nullable is Lib.StructWithStringProp notNull;
			return flag ? notNull.ToString() : "";
		}
	}
}
```

**Expected Behavior**:
error CS0165: Use of unassigned local variable 'notNull'

**Actual Behavior**:
The code compiles.

**Additional information**:
If you make any of the following changes, the code stops compiling and the expected error appears:
- replace struct's property with a field
- change struct's property type from `string` to, let's say, `int`
- move struct declaration from Lib to App

You can check out a solution that illustrates all these permutations and their effect on the compiler behavior [here](https://github.com/HellBrick/RoslynPatternMatchingDefiniteAssignmentIssue).

**Comment by jcouv** [comment]
I suspect this is a known issue due to the struct being empty. That said, I'm not sure why this would only affect an empty struct defined in metadata (as opposed to the same compilation/source).
Tagging @gafter to confirm.

**Comment by gafter** [comment]
This is an intentional behavior.  See also https://github.com/dotnet/roslyn/issues/30194 where we propose to fix this intentionally broken behavior.
